BaseApp/COD1218.txt

```diff
@@ -466,54 +466,54 @@ OBJECT Codeunit 1218 Map Incoming Doc to Purch Doc
 
         // Type
         FldNo := PurchaseLine.FIELDNO(Type);
-        EVALUATE(Type,GetValue(IntermediateDataImport,RecRef,FldNo,PurchaseLine.FIELDCAPTION(Type)));
+        EVALUATE(Type,GetValueFromIntermediate(IntermediateDataImport,RecRef,FldNo,PurchaseLine.FIELDCAPTION(Type)));
         ProcessField(TempProcessedLineFldId,RecRef,FldNo,FORMAT(Type));
 
         IF Type <> PurchaseLine.Type::" " THEN BEGIN
           // No.
           FldNo := PurchaseLine.FIELDNO("No.");
           ProcessField(TempProcessedLineFldId,RecRef,FldNo,
-            GetValue(IntermediateDataImport,RecRef,FldNo,PurchaseLine.FIELDCAPTION("No.")));
+            GetValueFromIntermediate(IntermediateDataImport,RecRef,FldNo,PurchaseLine.FIELDCAPTION("No.")));
 
           // Quantity
           FldNo := PurchaseLine.FIELDNO(Quantity);
           ProcessField(TempProcessedLineFldId,RecRef,FldNo,
-            GetValue(IntermediateDataImport,RecRef,FldNo,PurchaseLine.FIELDCAPTION(Quantity)));
+            GetValueFromIntermediate(IntermediateDataImport,RecRef,FldNo,PurchaseLine.FIELDCAPTION(Quantity)));
 
           // UOM
           FldNo := PurchaseLine.FIELDNO("Unit of Measure Code");
-          IF TryGetValue(IntermediateDataImport,RecRef,FldNo,PurchaseLine.FIELDCAPTION("Unit of Measure Code"),Value) THEN
+          IF TryGetValueFromIntermediate(IntermediateDataImport,RecRef,FldNo,PurchaseLine.FIELDCAPTION("Unit of Measure Code"),Value) THEN
             ProcessField(TempProcessedLineFldId,RecRef,FldNo,Value);
 
           // Direct Unit Cost
           FldNo := PurchaseLine.FIELDNO("Direct Unit Cost");
           ProcessField(TempProcessedLineFldId,RecRef,FldNo,
-            GetValue(IntermediateDataImport,RecRef,FldNo,PurchaseLine.FIELDCAPTION("Direct Unit Cost")));
+            GetValueFromIntermediate(IntermediateDataImport,RecRef,FldNo,PurchaseLine.FIELDCAPTION("Direct Unit Cost")));
         END;
 
         RecRef.MODIFY(TRUE);
       END;
     END;
 
-    LOCAL PROCEDURE GetValue@16(VAR IntermediateDataImport@1000 : Record 1214;RecRef@1003 : RecordRef;FieldID@1001 : Integer;FieldName@1002 : Text) : Text[250];
+    LOCAL PROCEDURE GetValueFromIntermediate@16(VAR IntermediateDataImport@1000 : Record 1214;RecRef@1003 : RecordRef;FieldID@1001 : Integer;FieldName@1002 : Text) : Text;
     VAR
-      Value@1004 : Text[250];
+      Value@1004 : Text;
     BEGIN
-      TryGetValue(IntermediateDataImport,RecRef,FieldID,FieldName,Value);
+      TryGetValueFromIntermediate(IntermediateDataImport,RecRef,FieldID,FieldName,Value);
 
       EXIT(Value);
     END;
 
     [TryFunction]
-    LOCAL PROCEDURE TryGetValue@20(VAR IntermediateDataImport@1000 : Record 1214;RecRef@1003 : RecordRef;FieldID@1001 : Integer;FieldName@1002 : Text;VAR Value@1004 : Text[250]);
+    LOCAL PROCEDURE TryGetValueFromIntermediate@20(VAR IntermediateDataImport@1000 : Record 1214;RecRef@1003 : RecordRef;FieldID@1001 : Integer;FieldName@1002 : Text;VAR Value@1004 : Text);
```

Comment:
You've now removed the length limitation of Value, but value is still passed into function ProcessField, which still only can handle Text[250] values. Is that by design?

Reply:
No must be a bug in my code that the test did not catch.

Reply:
Also. SetFieldValue further down truncates the value...

Reply:
ok, please fix 1218 and let's get this PR closed. Has been open WAY too long :)

Reply:
Looking at this now.  The Value in the TryGetValueFromIntermetidate is alway the Value field from the Intermetidate table so it is always max 250.  Updated COD1218

Reply:
That means COD1280 doesn't really have any chances (except that you've renamed GetValue to GetValueFromIntermediate, right)? I'd prefer not to do that change. Reverting that file.

Reply:
I renamed the function because we now have a the GetValue function in TAB1214 and can't have a function with the same name in COD1218